### PR TITLE
TPU training templates cleanup

### PR DESCRIPTION
**Pull request by dizcology**
For issue https://github.com/GoogleCloudPlatform/cloudml-samples/issues/326.

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/cloudml-samples/335)

### CODE REVIEW

**Comment by dizcology** [comment]
@gogasca the main files to review are `to_ipynb.py` and `parser_add.py`.  The rest are either automatically generated or the same source .py files with colab-specific code removed.

File: tpu/templates/tools/parser_add.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import glob
```
Comment (gogasca) [inline tpu/templates/tools/parser_add.py]: Add information about what this file does at the top.

File: tpu/templates/tools/parser_add.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        if node.value[0].name.value != 'parser' or node.value[1].name.value != 'add_argument':
+            continue
+
+        # args passed into the add_argument call
+        args = node.value[2].value
+
+        # get the arg name
+        assert args[0].target is None
+        arg_name = args[0].value.value.replace("'", '')
+        # print(arg_name)
```
Comment (gogasca) [inline tpu/templates/tools/parser_add.py]: Remove print ?

File: tpu/templates/tools/parser_add.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    # the `if __name__ == '__main__':` block
+    ifelseblock = red[-1]
+    nodes = ifelseblock.value[0]
+
+    for node in nodes:
+        # looking for those parser.add_argument calls
+        if node.type != 'atomtrailers':
+            continue
+
+        if node.value[0].name.value != 'parser' or node.value[1].name.value != 'add_argument':
```
Comment (gogasca) [inline tpu/templates/tools/parser_add.py]: Does node.value always have the same number of elements? (3) Not sure if there is a possible case which node.value[X] is inaccessible and may cause exception.

File: tpu/templates/tools/parser_add.py (hunk 2 above)
Comment (gogasca) [inline tpu/templates/tools/parser_add.py]: Remove print

File: tpu/templates/tools/parser_add.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        for arg in args[1:]:
+            if arg.target.value == 'help':
+                break
+
+        else:
+            # create a CallArgumentNode
+            arg = args[-1].copy()
+            arg.target.value = 'help'
+
+            # append
```
Comment (gogasca) [inline tpu/templates/tools/parser_add.py]: Not sure if # append comment is insightful or a better description is required.
https://github.com/google/styleguide/blob/gh-pages/pyguide.md

File: tpu/templates/tools/parser_add.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    '--save-checkpoints-steps': 'The number of training steps before saving each checkpoint.',
+    '--sequence-length': 'The sequence length for an LSTM model.',
+    '--gr-weight': 'The weight used in the gradient reversal layer.',
+    '--lambda': 'The trade-off between label_prediction_loss and domain_classification_loss.'
+}
+
+filenames = glob.glob('../**/trainer*.py')
+
+for filename in filenames:
+    with open(filename) as f:
```
Comment (gogasca) [inline tpu/templates/tools/parser_add.py]: do we need to specify open mode 'r' ?

File: tpu/templates/tools/parser_add.py (hunk 1 above)
Comment (dizcology) [reply tpu/templates/tools/parser_add.py]: glob is one of python's built in modules.

RedBaron is more problematic but to this point I haven't found a good substitute.

File: tpu/templates/tools/parser_add.py (hunk 2 above)
Comment (dizcology) [reply tpu/templates/tools/parser_add.py]: done

File: tpu/templates/tools/parser_add.py (hunk 4 above)
Comment (dizcology) [reply tpu/templates/tools/parser_add.py]: removed.

File: tpu/templates/tools/parser_add.py (hunk 5 above)
Comment (dizcology) [reply tpu/templates/tools/parser_add.py]: added.

File: tpu/templates/tools/parser_add.py (hunk 3 above)
Comment (dizcology) [reply tpu/templates/tools/parser_add.py]: Added a comment pointing to the documentation on the node structure.

**Pull request merged**